### PR TITLE
Feat notification refactor , uninstall and e2e tests

### PR DESCRIPTION
**Pull request by joshuef**
- Add notifications on failed install. 
- refactor notifications
- adds uninstall
- fixes mac uninstall
- updates testcafe to use prod app on CI
- fixes packaging issues

tracking #20

### CODE REVIEW

**Comment by joshuef** [comment]
Now closes #20

File: app/actions/alias/app_manager_actions.ts [hunk 1]
```diff
@@ -49,30 +51,43 @@ export const TYPES = {
...
+        } );
     } catch ( error ) {
         logger.error( error.message );
         const id: string = Math.random().toString( 36 );
 
         const errorNotification = {
             id,
             title: 'Remote application list could not be retrieved.',
-            notificationType: 'Native'
+            notificationType: 'standard'
```
Comment (manavbp) [inline app/actions/alias/app_manager_actions.ts]: I think this should be pushed to the notification constants file, as well as `i18n` integration is missing.

File: app/components/AppStateButton/AppStateButton.tsx [hunk 2]
```diff
@@ -125,23 +118,16 @@ export class AppStateButton extends React.Component<Props> {
                 <Button
```
Comment (manavbp) [inline app/components/AppStateButton/AppStateButton.tsx]: On retry install, according to the Figma designs, this should be a progress button to retry the installation!
Screenshot (145)

File: app/actions/alias/app_manager_actions.ts (hunk 1 above)
Comment (joshuef) [reply app/actions/alias/app_manager_actions.ts]: does i18n work outside of react?

File: app/components/AppStateButton/AppStateButton.tsx (hunk 2 above)
Comment (joshuef) [reply app/components/AppStateButton/AppStateButton.tsx]: ATM when paused, there's a button to resume the download. That's in line with what you're showing here.

Also please note, this PR expressly does not handle styling, just functionality.

File: app/actions/alias/app_manager_actions.ts (hunk 1 above)
Comment (joshuef) [reply app/actions/alias/app_manager_actions.ts]: seems like it does yep. It exports a helper func :+1:

File: app/actions/alias/app_manager_actions.ts [hunk 3]
```diff
@@ -49,30 +53,44 @@ export const TYPES = {
...
-        store.dispatch( setApps( apps.applications ) );
+
+        Object.keys( apps.applications ).forEach( ( theApp ) => {
+            store.dispatch( updateAppInfoIfNewer( apps.applications[theApp] ) );
+        } );
     } catch ( error ) {
         logger.error( error.message );
         const id: string = Math.random().toString( 36 );
 
         const errorNotification = {
```
Comment (manavbp) [inline app/actions/alias/app_manager_actions.ts]: I think we should maybe push this to the notification's constant file

**Comment by manavbp** [comment]
Also, I think later on we could maybe try making user `redux-promise-middleware` ( https://github.com/pburtchaell/redux-promise-middleware/tree/b8e4b0ac33efa726303738576da16c652cc8950e ) to help simplify the actions we have at the moment. Other than that the PR looks good to me @joshuef, other than the one comment to maybe add the custom notification's to the notification constants file.

**Comment by manavbp** [comment]
Note: Download and install does not work for Windows at the moment. Will create an issue for it once this PR is in!

File: app/actions/alias/app_manager_actions.ts (hunk 3 above)
Comment (joshuef) [reply app/actions/alias/app_manager_actions.ts]: Ah, I'm not sure that file actually _serves_ a purpose... We're just moving these errors, triggered in specific circumstances one file away from where they are triggered. I'm not sure that abstractions servers a purpose when the messages etc are `I18n`...

Comment (manavbp) [reply app/actions/alias/app_manager_actions.ts]: Okay that make's sense to me as well

**Comment by joshuef** [comment]
I actually looked at that @manavbp (redux-promise-middleware), but we're not doing a simple promise here, we're just starting a cascade here. 
I agree the action / boilerplate would be cleaner/ nicer.
But configuring it to _be_ a promise, across processes etc seemed like a lot of work for very little gain.

**Pull request merged**